src/network/message.rs

```diff
@@ -250,28 +258,18 @@ impl<N: Network, E: Environment> Encoder<Message<N, E>> for Message<N, E> {
     type Error = anyhow::Error;
 
     fn encode(&mut self, message: Message<N, E>, dst: &mut BytesMut) -> Result<(), Self::Error> {
-        // Serialize the message into a buffer.
-        let buffer = message.serialize()?;
+        // Prepare the room for the length of the payload.
+        dst.extend_from_slice(&0u32.to_le_bytes());
 
-        // Ensure the message does not exceed the maximum length limit.
-        if buffer.len() > E::MAXIMUM_MESSAGE_SIZE {
-            return Err(std::io::Error::new(
-                std::io::ErrorKind::InvalidData,
-                format!("Frame of length {} is too large.", buffer.len()),
-            )
-            .into());
-        }
```

Comment:
this check is not needed for outbound messages - it is only required for the inbound ones (as those are the only ones that can be maliciously crafted to be too long)